client/packages/system/src/Patient/JsonForms/components/BMI.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from 'react';
+import {
+  composePaths,
+  ControlProps,
+  rankWith,
+  uiTypeIs,
+} from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import {
+  DetailInputWithLabelRow,
+  FORM_LABEL_WIDTH,
+  NumUtils,
+} from '@openmsupply-client/common';
+
+export const bmiTester = rankWith(10, uiTypeIs('BMI'));
+
+const round = (value: number) => Math.round(value * 100) / 100;
+
+const UIComponent = (props: ControlProps) => {
+  const { data, handleChange, label, path } = props;
+  const [bmi, setBmi] = useState<number | null>(null);
+
+  useEffect(() => {
+    if (!data) return;
+
+    const height = NumUtils.parseString(data.height);
+    const weight = NumUtils.parseString(data.weight);
+
+    if (!height || !weight) return;
+
+    setBmi(round(weight / (height * height)));
```

Comment:
is there any need for bmi state ?

Reply:
no - I've changed my mind about it now.. 
have just updated the data directly.